src/domain/buildingBlocks/Pluto.ts

```diff
@@ -121,76 +96,6 @@ export interface Pluto {
    */
   getDIDPrivateKeysByDID(did: DID): Promise<Array<PrivateKey>>;
 
-  /**
-   * Retrieve private key for a given key ID.
-   */
-  getDIDPrivateKeyByID(id: string): Promise<PrivateKey | null>;
-
-  /**
-   * Retrieve all stored DID pairs (DIDComm connections).
-   */
-  getAllDidPairs(): Promise<Array<DIDPair>>;
-
-  /**
-   * Retrieve a DID pair containing a given DID as either host or receiver.
-   */
-  getPairByDID(did: DID): Promise<DIDPair | null>;
-
-  /**
-   * Retrieve a DID pair by a given pair name.
-   */
-  getPairByName(name: string): Promise<DIDPair | null>;
-
-  /**
-   * Retrieve all stored DIDComm messages.
-   */
-  getAllMessages(): Promise<Array<Message>>;
-
-  /**
-   * Retrieve all stored DIDComm messages, received from or sent to a given DID
-   */
-  getAllMessagesByDID(did: DID): Promise<Array<Message>>;
-
-  /**
-   * Retrieve all stored, sent DIDComm messages.
-   */
-  getAllMessagesSent(): Promise<Array<Message>>;
-
-  /**
-   * Retrieve all stored, received DIDComm messages.
-   */
-  getAllMessagesReceived(): Promise<Array<Message>>;
-
-  /**
-   * Retrieve all stored DIDComm messages, sent to a given DID.
-   */
-  getAllMessagesSentTo(did: DID): Promise<Array<Message>>;
-
-  /**
-   * Retrieve all stored DIDComm messages, received from a given DID.
-   */
-  getAllMessagesReceivedFrom(did: DID): Promise<Array<Message>>;
-
-  /**
-   * Retrieve all stored DIDComm messages with given message type, and
-   * optionally, related to a given DID. "Related" means that message should
-   * contain a given DID in either "from" or "to" field.
-   */
-  getAllMessagesOfType(
-    type: string,
-    relatedWithDID?: DID
-  ): Promise<Array<Message>>;
-
-  /**
-   * Retrieve all DIDComm messages containing given "from" AND "to" DIDs.
-   */
-  getAllMessagesByFromToDID(from: DID, to: DID): Promise<Array<Message>>;
-
-  /**
-   * Retrieve a DIDComm message by ID.
-   */
-  getMessage(id: string): Promise<Message | null>;
```

Comment:
we never retrieve messages from Pluto.
ConnectionsManager retrieves new messages via didcomm, then stores and emits them.

Reply:
Ok but this is also because the DEMO's are maybe not listing how many credentials we will show. @dorultan is for example working on a task that will need the credential list soon. So i'd say, at least a fetch all and a fetch by ID would be needed.